rte-kernels/api/rte_types.h.in

```diff
@@ -0,0 +1,28 @@
+/* This code is part of Radiative Transfer for Energetics (RTE)
+
+Contacts: Makepath LLC
+email:  rrtmgp@aer.com
+
+Copyright 2024-
+   Trustees of Columbia University in the City of New York
+   All right reserved.
+
+Use and duplication is permitted under the terms of the
+    BSD 3-clause license, see http://opensource.org/licenses/BSD-3-Clause
+
+This `.h` file defines C-compatible Boolean and floating-point types.
+  See mo_rte_type.F90 for the Fortran equivalent.
+  Adapted from code written by Chiel van Heerwaarden at Wageningen University and Research
+
+*/
+#pragma once
+
+#include <cfloat.h>  // Ensure FLT_EPSILON and DBL_EPSILON are available
+#include <stdbool.h>
+
+// Define a C-compatible Boolean type
+typedef @RTE_BOOL@ Bool;
```

Comment:
If I understand this correctly, the resulting header pollutes the user's code. Imagine they already have `Bool` somewhere in the code that means a different thing. If you need to define a type (for reasons that are not clear to me), it should have a less generic name (e.g. `rte_bool`).

I also don't understand why it has to be `@RTE_BOOL@` and not simply `_Bool`.

Reply:
Also, this is not a C++ header anymore as the title says. A C++ compiler might not know `_Bool` and should include C headers differently: instead of `cstdbool.h`, it should be `<cstdbool>`.